File: exercises/java/src/main/java/checkout/Checkout.java
package checkout;

class Checkout {
    private int total;
    private int numberOfA = 0;
    private int numberOfB = 0;
    private Receipt receipt = new Receipt();

    void scan(String sku) {
        if ("A".equals(sku)) {
            if (++numberOfA % 3 == 0) {
                total += 30;
            }
            else {
                total += 50;
            }
            receipt.scannedA();
        } else if ("B".equals(sku)) {
            if (++numberOfB % 2 == 0) {
                total += 15;
            }
            else {
                total += 30;
            }
            receipt.scannedB();
        } else if ("C".equals(sku)) {
            total += 20;
            receipt.scannedC();
        } else if ("D".equals(sku)) {
            total += 15;
            receipt.scannedD();
        }
    }

    int total() {
        return total;
    }

    String receipt() {
        return receipt.text();
    }
}

File: exercises/c-sharp/apprentice-bootcamp-fundamentals-2/apprentice-bootcamp-fundamentals-2/Checkout.cs
namespace apprentice_bootcamp_fundamentals_2 {
  public class Checkout {
    private int _total;
    private int _numberOfA;
    private int _numberOfB;
    private Receipt _receipt = new Receipt();

    public string Receipt {
      get => this._receipt.Text;
    }

    public int Total {
      get => this._total;
    }

    public void Scan(string sku) {
      if ("A".Equals(sku)) {
        this._total += 50;
        this._receipt.ScannedA();
        this._numberOfA++;
        if (this._numberOfA % 3 == 0) {
          this._total -= 20;
        }
      } else if ("B".Equals(sku)) {
        this._total += 30;
        this._receipt.ScannedB();
        this._numberOfB++;
        if (this._numberOfB % 2 == 0) {
          this._total -= 15;
        }
      } else if ("C".Equals(sku)) {
        this._total += 20;
        this._receipt.ScannedC();
      } else if ("D".Equals(sku)) {
        this._total += 15;
        this._receipt.ScannedD();
      }

    }
  }
}